### ISSUE TITLE
FPN-7 | Phase 2 | Train Baseline XGBoost Classifier

### ISSUE THREAD

**Issue by LavanyaSrinivas06** [LavanyaSrinivas06/AI-First-Preauth-Fraud-Detection#56]
### Ticket ID: FPN-7  
Phase: Phase 2 — Model Training & Evaluation  
Branch Name: feature/FPN-7-train-xgboost

---

### Context
The supervised XGBoost classifier serves as the baseline model for detecting known fraudulent transactions.  
This ticket involves training, hyperparameter tuning, and metric evaluation on the processed dataset to establish benchmark performance.

---

### Scope
- Train XGBoost classifier on balanced training data.  
- Tune hyperparameters using grid search or randomized search.  
- Evaluate performance using precision, recall, F1-score, and ROC-AUC.  
- Save trained model and metric report.

---

### Steps

1. **Create training script**
   - File: `scripts/train_xgboost.py`
   - Load data from `data/processed/train.parquet`, `val.parquet`, `test.parquet`.
   - Separate features and target (`Class`).
   - Import and initialize:
     ```python
     from xgboost import XGBClassifier
     model = XGBClassifier(
         n_estimators=300,
         learning_rate=0.05,
         max_depth=6,
         subsample=0.8,
         colsample_bytree=0.8,
         random_state=42,
         scale_pos_weight=ratio  # adjust for imbalance
     )
     ```

2. **Parameter tuning**
   - Use scikit-learn’s `GridSearchCV` or `RandomizedSearchCV` on parameters such as:
     - `max_depth`, `learning_rate`, `subsample`, `colsample_bytree`.
   - Evaluate using F1-score.

3. **Model evaluation**
   - Predict on validation and test sets.
   - Compute:
     - Precision, Recall, F1-score, ROC-AUC.
     - Confusion matrix and classification report.
   - Save metrics to JSON:
     ```json
     artifacts/xgb_metrics.json
     ```

4. **Persist model**
   - Save with joblib:
     ```python
     import joblib
     joblib.dump(model, "artifacts/xgb_model.pkl")
     ```

5. **Visualization (optional)**
   - Plot feature importance using `xgboost.plot_importance`.
   - Save to `artifacts/plots/xgb_feature_importance.png`.

6. **Documentation**
   - Create `docs/xgb_training_report.md` summarizing:
     - Data used, parameters tested, final scores, observations.

---

### Acceptance Criteria
- [ ] Model successfully trains without errors.  
- [ ] Achieves baseline F1 ≥ 0.85 on validation data.  
- [ ] Metrics file (`xgb_metrics.json`) saved.  
- [ ] Model file (`xgb_model.pkl`) saved.  
- [ ] Report (`xgb_training_report.md`) includes parameters and metrics.

---

### Artifacts
- `scripts/train_xgboost.py`  
- `artifacts/xgb_model.pkl`  
- `artifacts/xgb_metrics.json`  
- `docs/xgb_training_report.md`

---

### Helpful Links
- XGBoost documentation: https://xgboost.readthedocs.io/  
- Scikit-learn model evaluation metrics: https://scikit-learn.org/stable/modules/classes.html#module-sklearn.metrics

---

### Effort Estimate
M (1–2 days)
